### PR TITLE
[RFC] Multiline {## Comment Block ##} 

### PR DESCRIPTION
**Pull request by smnandre**
I started to work on this feature, following @ThomasLandauer  comment in #3951 (and previously discussed in #2897)

So this PR is a starting point to discuss implementation, any particularities, specific behaviours, etc.

The idea is that anything between  `{##` and `##}`  would be parsed as a comment (and thus ignored)

```twig
Hello {## foo bar 
Foo
Bar {# Bar #}
Foo ##} Hello
```

would render

```html
Hello  Hello
```

### CODE REVIEW

File: tests/LexerTest.php [hunk 1]
```diff
@@ -131,6 +131,33 @@ public function testLongComments()
...
+        $template = '{## {# '.str_repeat('*', 100000).' #} ##}';
+
+        $lexer = new Lexer(new Environment($this->createMock(LoaderInterface::class)));
+
+        $res = $lexer->tokenize(new Source($template, 'index'));
+        $this->assertTrue($res->isEOF());
+
+        // add a dummy assertion here to satisfy PHPUnit, the only thing we want to test is that the code above
+        // can be executed without throwing any exceptions
+        $this->addToAssertionCount(1);
```
Comment (stof) [inline tests/LexerTest.php]: no need for a dummy assertion here, as there is an actual assertion just above.

File: tests/LexerTest.php [hunk 2]
```diff
@@ -178,6 +205,49 @@ public function testBigNumbers()
...
+            ."baz\n"
+            ."}}\n##}\n";
+
+        $lexer = new Lexer(new Environment($this->createMock(LoaderInterface::class)));
+        $stream = $lexer->tokenize(new Source($template, 'index'));
+
+        $this->assertTrue($stream->isEOF());
+    }
+
+    public function testMultiLineCommentNestedBlock()
```
Comment (stof) [inline tests/LexerTest.php]: this test does not seem to have any nested block.

File: tests/LexerTest.php [hunk 3]
```diff
@@ -178,6 +205,49 @@ public function testBigNumbers()
...
+
+        $lexer = new Lexer(new Environment($this->createMock(LoaderInterface::class)));
+        $stream = $lexer->tokenize(new Source($template, 'index'));
+
+        $stream->test(Token::TEXT_TYPE, 'baz');
+        $stream->next();
+        $this->assertTrue($stream->isEOF());
+    }
+
+    public function testMultiLineCommentBlockWithInlineBLocks()
```
Comment (stof) [inline tests/LexerTest.php:231]: this seems to be about inline comments

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by smnandre** [comment]
Thank you for this first feedback @stof

**Comment by smnandre** [comment]
Is this something really needed / asked for ? 

Just to be sure it is worth the time needed to refine this PR, add documentation, think how to handle BC, ....

**Comment by fabpot** [comment]
As far as I know, this syntax is not supported by any other language similar to Twig (Jinja and the like), so I'm not sure if the use cases are worth the added complexity in core.

**Comment by smnandre** [comment]
Well let's close then :)

**Comment by c33s** [comment]
i would have been happy to see this feature finally. it helps to easily comment out blocks of text which have comments inside and preserving them after you uncomment the block again.

**Pull request closed without merging**